pkgs/build-support/writers/default.nix

```diff
@@ -225,11 +225,12 @@ let
   writeNginxConfig = name: text: pkgs.runCommandLocal name {
     inherit text;
     passAsFile = [ "text" ];
-    nativeBuildInputs = [ gixy ];
+    nativeBuildInputs = [ gixy nginx ];
   } /* sh */ ''
     # nginx-config-formatter has an error - https://github.com/1connect/nginx-config-formatter/issues/16
     awk -f ${awkFormatNginx} "$textPath" | sed '/^\s*$/d' > $out
     gixy $out
+    nginx -c $out -e /dev/null -t
```

Comment:
Would this prevent config errors from being shown in the derivation build output? Perhaps you meant the behavior of something like `-q` instead of `-e /dev/null`?

Reply:
As far as I can tell, it only sets the log file. It will still output to stdout. It tried opening `/var/log` without it but I tried that with a broken nginx config and it works as intended.

Reply:
Ohh I see the issue. Yeah that might be a problem for access logs and stuff too, wouldn't it?

Reply:
Nope, it just needs to know if the file exists lol

Reply:
Is there a flag to let you only do the syntax test part? (Or maybe exit code is different?)

It would also be a problem for includes and such too

Reply:
There's not, the only flags related to testing are the regular test (`-t`) and the test and dump config (`-T`).

Reply:
Maybe we need to use another syntax checker like https://github.com/nginxinc/crossplane instead of `nginx -t` then. Let me see if it would have caught the issue in the discord or not

Reply:
Yeah I think the best we're going to get is `crossplane parse --single-file $out | jq '.status != "ok" and (.errors | map(.error) | join("\n") + "\n" | halt_error) // empty'` (apologies for probably inefficient `jq` expression, I am no `jq` expert)

It doesn't catch the `server_name *;` example from the Discord, but that does catch more than gixy does (like bad syntax), so maybe that's worth it

There is a `--strict` option for crossplane but I worry if people have custom modules it might get upset about that in strict mode so probably best to leave it out or make it opt-in for the writer

Reply:
Maybe `*` is supposed to work but the docs doesn't say and it's broken in nginx? Or they just forgot to write the check in `nginx`.

Reply:
Well, nginx checks it. But `crossplane` doesn't though (maybe should make an upstream issue about that?)

Reply:
Yeah, should let them know.

Reply:
Since I really don't think we can reliably `nginx -t` (and that check runs at runtime anyway before starting/reloading the service), I think `crossplane` is probably the best option we have for syntax checking at build-time

Reply:
I changed it to use crossplane now.